### PR TITLE
Mini challenge 3

### PR DESCRIPTION
**Pull request by vmcruz**
#### Acceptance Criteria

- [x] Videos in the Home View are fetched from the YouTube API and displayed correctly.
- [x] Users can search for YouTube videos using the search field on the header.
- [x] A video can be played from the Video Details View after clicking on it.
- [x] The video information and related videos list are displayed correctly on the Video Details View.
- [x] When a user clicks on a related video the video player, information and related videos list are updated.

#### Bonus Points

- [x] Custom Hooks for API calls are implemented and tested correctly.
- [ ] Testing coverage is above 60%. (Please include a screenshot of the code coverage report in your PR description).

#### Screenshots from coverage
Screen Shot 2021-03-12 at 13 42 24
Screen Shot 2021-03-12 at 13 42 13
Screen Shot 2021-03-12 at 13 42 17

### CODE REVIEW

File: src/api/__test__/ytAPI.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    status: 403,
+  };
+
+  beforeEach(() => {
+    global.fetch = jest.fn().mockResolvedValue(successMock);
+  });
+
+  afterEach(() => {
+    global.fetch.mockRestore();
+  });
```
Comment (PacoMojica) [inline src/api/__test__/ytAPI.test.js:21]: This test is really good, I like the setup, the cases, the assertions, overall a well written test 👍

File: src/api/ytAPI.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  throw new Error(
+    `Error ${response.status || ''}: An error ocurred while searching for [${
+      searchTerm.type
+    }: ${searchTerm.value}].`
+  );
+}
+
+function searchQuery({ searchTerm, pageToken }) {
+  if (!searchTerm) return null;
+
```
Comment (PacoMojica) [inline src/api/ytAPI.js:22]: Although these functions are not hooks, the way you abstracted the logic makes things easy to reuse

File: src/components/Overlay/Overlay.component.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    onDismiss();
+  }
+
+  useEffect(() => {
+    if (!scroll) {
+      document.body.style.overflow = 'hidden';
+    }
+  }, [scroll]);
+
+  async function handleKeyUp(e) {
```
Comment (PacoMojica) [inline src/components/Overlay/Overlay.component.jsx:21]: I was not expecting the video to show in a responsive modal but it looks good

File: src/components/Overlay/__test__/Overlay.test.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  it('renders with default props', () => {
+    const { getByTestId } = render(<Overlay {...props} />);
+
+    const overlay = getByTestId('overlay');
+
+    expect(overlay).toBeInTheDocument();
+  });
+
+  it('has a close option', () => {
```
Comment (PacoMojica) [inline src/components/Overlay/__test__/Overlay.test.jsx:31]: You are covering several edge cases for your components, which is a good thing, I guess thats why you have a high branch percentage

File: src/hooks/useYoutubeSearch.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  const { getItem, setItem } = useCache({ ttl: 3600 });
+  const items = useRef([]);
+  const [searchTerm, setSearchTerm] = useState(null);
+  const [pagination, setPagination] = useState({});
+  const [error, setError] = useState(null);
+
+  const youtubeSearch = useCallback(
+    async ({ pageToken } = {}) => {
+      try {
+        const cacheKey = `${cachePrefix}@${searchTerm}${
```
Comment (PacoMojica) [inline src/hooks/useYoutubeSearch.js:16]: Ohh, so here is where you use the functions defined in `api/ytAPI`, awesome. I think that once you add the tests for `useCache`, `useDebouncer` and `useYoutubeSearch` the test coverage will increase considerably

**Review by PacoMojica — COMMENTED** [review]
Hi Victor, your deliverable is great, you have everything that was required up to this point, as you mentioned the only thing that is missing is the 60% test coverage, but that is not a problem, the tests that you have are well defined. Next time submit your progress, if you are missing something you can include it in the next delivery. Good job 👍

**Review by PacoMojica — APPROVED** [review]
